locations/storefinders/where2getit.py

```diff
@@ -161,3 +164,31 @@ def parse_locations(self, response, **kwargs):
 
     def parse_item(self, item: Feature, location: dict, **kwargs):
         yield item
+
+    def storefinder_exists(response: Response) -> bool | Request:
+        # Example: https://www.att.com/stores/
+        # <script>
+        #     let locationsCount;
+        #     var W2GI = {
+        #         config: {
+        #             appkey: '...',
+        if response.xpath('//script[contains(text(), "W2GI")]').get():
+            return True
+
+        # https://locations.vans.com/index.html
+        # <script type="text/javascript" src="/w2gi/javascript/ace/2.1/W2GI_core.js"></script>
+        if response.xpath('//script[contains(@src, "W2GI")]').get():
+            return True
+
+        # https://www.dollartree.com/store-locator
+        # https://www.familydollar.com/store-locator
+        if response.xpath('//script[contains(text(), "ccstoreui")]').get():
+            return True
+
+        return False
+
+    def extract_spider_attributes(response: Response) -> dict | Request:
+        return {
+            "allowed_domains": [urlparse(response.url).netloc],
+            "api_key": "Inspect browser console to determine",
```

Comment:
From https://www.mattressfirm.com/en-us/stores/ we could extract the key from (perhaps with a regular expression) `appkey: '[A-F0-0]{8}-[A-F0-0]{4}-[A-F0-0]{4}-[A-F0-0]{4}-[A-F0-0]{12}',`

Reply:
I saw unfortunately a few different presentations.

Works for:
```
        if response.xpath('//script[contains(text(), "var W2GI = {")]').get():
            return True
```

```
        # https://locations.vans.com/index.html
        # <script type="text/javascript" src="/w2gi/javascript/ace/2.1/W2GI_core.js"></script>
        if response.xpath('//script[contains(@src, "W2GI")]').get():
            return True
```
Has it as ```            <appkey>CFCAC866-ADF8-11E3-AC4F-1340B945EC6E</appkey>```

But then:
```
        # https://www.dollartree.com/store-locator
        # https://www.familydollar.com/store-locator
        if response.xpath('//script[contains(text(), "ccstoreui")]').get():
            return True
```
Has it as:
```https://api.slippymap.com/rest?&xml_request=%3Crequest%3E%0A%3Cappkey%3E57B1E48E-D096-11E8-94D9-9926407E493E%3C%2Fappkey%3E%0A%3Cformdata%20id%3D%22locatorsearch%22%3E%0A%3Cdataview%3Estore_default%3C%2Fdataview%3E%0A%3Climit%3E1%3C%2Flimit%3E%0A%3Cgeolocs%3E%0A%3Cgeoloc%3E%0A%3Caddressline%3E%3C%2Faddressline%3E%0A%3Clongitude%3E138.5218451%3C%2Flongitude%3E%0A%3Clatitude%3E-34.8493422%3C%2Flatitude%3E%0A%3C%2Fgeoloc%3E%0A%3C%2Fgeolocs%3E%0A%3Csoftmatch%3E1%3C%2Fsoftmatch%3E%0A%3Cwhere%3E%3Cname%3E%3Cilike%3EFAMILY%20DOLLAR%20%23%3C%2Filike%3E%3C%2Fname%3E%3C%2Fwhere%3E%0A%3Cproximitymethod%3Estraightline%3C%2Fproximitymethod%3E%0A%3Csearchradius%3E10%7C25%7C50%7C75%7C100%7C125%7C200%3C%2Fsearchradius%3E%0A%3C%2Fformdata%3E%0A%3C%2Frequest%3E```

It's sort of three-storefinders-in-one which makes it ugly.

Reply:
I see what you mean. It's far from a standard.

I suggested in #7530 instead that we should rethink XPath extraction and instead have the ability to intercept requests to API endpoints using Playwright, and take API keys out of these intercepted requests. I think this may be the best generic approach from now on to tackle most of the store finder detection / API key extraction where there is a single or small list of known API endpoints we can watch for and intercept calls to.

Reply:
I guess, what's the outcome? This is messier than it really needs to be, I'm happy to just leave it on my branch for a while but close this PR until a better method is available.